pennylane/ops/op_math/symbolicop.py

```diff
@@ -166,10 +166,24 @@ class ScalarSymbolicOp(SymbolicOp):
     def __init__(self, base, scalar: float, id=None):
         self.scalar = np.array(scalar) if isinstance(scalar, list) else scalar
         super().__init__(base, id=id)
-        self._batch_size = self._check_and_compute_batch_size(scalar)
+        self._batch_size = _UNSET_BATCH_SIZE
 
     @property
     def batch_size(self):
+        if self._batch_size is _UNSET_BATCH_SIZE:
+            base_batch_size = self.base.batch_size
+            if qml.math.ndim(self.scalar) == 0:
+                # coeff is not batched
+                self._batch_size = base_batch_size
+                return base_batch_size
+            # coeff is batched
+            scalar_size = qml.math.size(self.scalar)
+            if base_batch_size is not None and base_batch_size != scalar_size:
+                raise ValueError(
+                    "Broadcasting was attempted but the broadcasted dimensions "
+                    f"do not match: {scalar_size}, {base_batch_size}."
+                )
+            self._batch_size = scalar_size
```

Comment:
Fine by me to leave this as is, but one could also consider to _always_ return via the last line and put this part in an `else` clause, just to simplify the overall control flow logic.

Reply:
changed to be in an `else` block

Reply:
Thanks! Both ways would be okay, I just found myself slightly confused for a moment, _although_ I prefer return-early usually :sweat_smile: